source/extensions/filters/udp/udp_proxy/hash_policy_impl.cc

```diff
@@ -20,13 +21,34 @@ class SourceIpHashMethod : public HashPolicyImpl::HashMethod {
   }
 };
 
+class KeyHashMethod : public HashPolicyImpl::HashMethod {
+public:
+  explicit KeyHashMethod(const std::string& key) : key_(key) {}
+
+  absl::optional<uint64_t>
+  evaluate(const Network::Address::Instance& downstream_addr) const override {
+    UNREFERENCED_PARAMETER(downstream_addr);
+    if (!key_.empty()) {
+      return HashUtil::xxHash64(key_);
+    }
+
+    return absl::nullopt;
+  }
+
+private:
+  const std::string& key_;
```

Comment:
Make sure the string this member is referencing doesn't go out of scope. This may be the reason why the CI test if failing.
And speaking of the string... Wouldn't it be more efficient to calculate the hash only once upon the hash method's instantiation if `key` is `const` anyway?

Reply:
> Make sure the string this member is referencing doesn't go out of scope. This may be the reason why the CI test is failing.

It was the problem, thanks for the hint. 

> And speaking of the string... Wouldn't it be more efficient to calculate the hash only once upon the hash method's instantiation if `key` is `const` anyway?

I changed the `KeyHashMethod` class as you suggested. I have a question.  According to the doc string, `key` must have a minimum length of one. I'm not sure if I should check whether `key` is empty or not in the constructor (`explicit KeyHashMethod(const std::string& key) { hash_ = HashUtil::xxHash64(key); }`). I feel like I should, but as I see it would be unnecessary. WDYT?

Reply:
Yes, it's not necessary indeed - protobuf will validate the value. But it won't harm to `ASSERT()` the key is not empty to make sure the tests are properly set up.

Since `HashUtil::xxHash64()` handles empty inputs gracefully you can move `hash_` initialization to the init list, e.g.:
```c++
explicit KeyHashMethod(const std::string& key) : hash_{HashUtil::xxHash64(key)} { ASSERT(!key.empty()); }
```